### ISSUE TITLE
pyenv does not respect aria2.conf

### ISSUE THREAD

**Issue by herrbischoff** [pyenv/pyenv#1126]
## Issue

On a FreeBSD server, I had the problem of installation via pyenv failing due to not being able to download the Python source code. I found out that the `aria2c` utility is used if it is installed. However, it does not use the settings in the config file, necessitating the following command line to successfully install:

```
env PYTHON_BUILD_ARIA2_OPTS="--ca-certificate=/etc/ssl/cert.pem" pyenv install 3.6.3
```

## Solution

On FreeBSD, the SSL certificates are not located at `/etc/ssl/certs/ca-certificates.crt`, which is a Linuxism that should be avoided. This should be fixed in `aria2c`. However, `pyenv` should either respect the local config settings of `aria2c` or remove support for it. The reason for failure is not obvious and only visible when using the verbose flag (`-v`).

**Comment by joshfriend** [comment]
seems like that is intentional #625

**Comment by herrbischoff** [comment]
Thanks for the hint. But wow, this is kind of shortsighted. I can understand not using the user configuration. But since this can be problematic, using `aria2c` should really be optional, in a way that it requires to be explicitly enabled. Possibly through an env variable like `PYTHON_BUILD_ARIA2_ENABLE`. Both curl and wget are well-behaved and not affected by any of this.

**Comment by joshfriend** [comment]
If aria2 allows it's config file to override the `-o` command line flag, what else can we do?

aria2 offers features that wget and curl don't which is why it is the default (#534)

**Comment by yyuu** [comment]
With current implementation, it'll always pick up the first found preferred HTTP client on the system. And, `aria2c` is at the top.

https://github.com/pyenv/pyenv/blob/v1.2.2/plugins/python-build/bin/python-build#L331-L345

Introducing some new environment variable like `PYTHON_BUILD_HTTP_CLIENT="curl"` might help users to deal with the situation like `aria2c` on FreeBSD. As a happy side effect, it might help simplifying some `bats` tests of `python-build` for most cases.

https://github.com/pyenv/pyenv/blob/v1.2.2/plugins/python-build/test/build.bats#L11

**Comment by joshfriend** [comment]
I like the `PYTHON_BUILD_HTTP_CLIENT` idea :+1:

**Comment by herrbischoff** [comment]
Sounds like a reasonable idea, although I’m not a fan of software that tries to be too clever for its own good.

**Comment by yyuu** [comment]
I've merged #1146 into master. With using the latest revision, now `PYTHON_BUILD_HTTP_CLIENT=curl` will let python-build to use the preferred HTTP client on your system.

**Comment by salotz** [comment]
This isn't documented anywhere. I was having issues with using curl needing to be configured to go through a proxy not working. `wget` does respect the config file.
